### DISCUSSION TITLE
Introduction of uniGradICON into MONAI

### DISCUSSION THREAD

**Discussion by HastingsGreer** [Project-MONAI/MONAI#8236] [Ideas]
Hi! I am a developer of https://github.com/uncbiag/uniGradICON, a pretrained neural network for image registration that generalizes to new registration tasks. I think it may be a good fit to bring into the MONAI project, as it fills several holes that medical developers need:

For segmentation, MONAI users have access to pretrained models, allowing them to leverage NVIDIA's training resources. However, for registration, MONAI lacks models that generalize to user data. uniGradICON (and multiGradICON for multimodal registration) would allow users to perform registration while staying in the MONAI ecosystem, without having to take on model training and hyperparameter tuning.

Spacing, origin, and direction matrices are particularly important for registration tasks. This is currently a pain point for MONAI (for example, #8234 #6313) and uniGradICON solves this by providing functionality to directly create an ITK transform in physical coordinates, given two input images with correct spatial metadata. This could either be taken into MONAI as part of the uniGradICON tool, or brought in as an independent element for use with GlobalNet and LocalNet.

uniGradICON is build with components from the icon_registration library, which I have began porting to the MONAI idiom here: https://github.com/Project-MONAI/research-contributions/pull/361/files#diff-022eb283edf5aad9a215774ec19ad2ac12d9786ae7930abb2df3a61a1952bf19

Talking with Stephen, this may be a better fit for monai core than for research contributions. I am excited to hear your thoughts!

--Hastings

**Comment by KumoLiu** [comment]
Hi @HastingsGreer, you can refer to the contribution guide here: https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md
And if you could split thing into several prs, that would be great. Thanks!

**Comment by aylward** [comment]
This would be an outstanding contribution to MONAI.   Registration is very much needed for medical applications, and a good software foundation is critical.   Seems like the foundation Hastings has put together is quite good - creating transforms in physical space (rather than in image space) so that those transforms can then be applied to geometries, other images, etc etc.    Very nice!!!

I think this belongs in core!

There is documentation in ITK and SimpleITK on why registration should be in physical space.   See also this CMU lecture: https://www.cs.cmu.edu/~galeotti/methods_course/ITK_Registration.pdf

@Nic-Ma , @SachidanandAlle , @jjomier , @ericspod , @diazandr3s
